src/main/java/study/core/Point.java

```diff
@@ -0,0 +1,29 @@
+package study.core;
+
+public class Point {
+    private final int position;
+    private final Edge edge;
+
+    public Point(int position, Edge edge) {
+        this.position = position;
+        this.edge = edge;
+    }
+
+    public Edge getEdge() {
+        return edge;
+    }
+
+    public boolean isCurrent() {
+        return edge.isCurrent();
+    }
+
+    public int move() {
+        if (edge.move() == Direction.RIGHT) {
```

Comment:
Nit: 객체에게 메시지를 보내서 처리해봐도 좋을 것 같습니다.
```suggestion
        if (edge.isRight()) {
```